### PR TITLE
Fix gaussian basis set

### PR DESCRIPTION
**Pull request by ladinesa**
A parsing error for gaussian has been reported for this [upload](https://nomad-lab.eu/prod/v1/gui/user/uploads/upload/id/XenO4Le6QyOHXHBpdxtVgA) . The basis set cannot be resolved so I simply set it to None, but @ndaelman-hu you should know more.

The method normalizer also fails so I put up a fix in nomad.

### CODE REVIEW

**Comment by JosePizarro3** [comment]
Hi there,

I cannot see this upload, but is this the last one by Akseli Mansikkamäki (https://nomad-lab.eu/prod/v1/gui/upload/id/MASTxxsySg-SkFffP9j2Zw)? Do you know details on the data? I think he is studying magnetism in charged molecules, but I'd like to make sure.

**Comment by ndaelman-hu** [comment]
> Hi there,
> 
> I cannot see this upload, but is this the last one by Akseli Mansikkamäki (https://nomad-lab.eu/prod/v1/gui/upload/id/MASTxxsySg-SkFffP9j2Zw)? Do you know details on the data? I think he is studying magnetism in charged molecules, but I'd like to make sure.

Neither I can see the upload. @ladinesa Could you ask permissions to share the raw files or have us included?

File: electronicparsers/gaussian/parser.py [hunk 1]
```diff
@@ -1414,7 +1415,7 @@ def resolve_xc_functional(parameter):
             BasisSetContainer(
                 type='atom-centered orbitals',
                 scope=['wavefunction'],
-                basis_set=[bs],
+                basis_set=[bs] if bs is not None else [],
             )
         ]
```
Comment (ndaelman-hu) [inline electronicparsers/gaussian/parser.py]: You can leave out the entire `BasisSetContainer` if the requirement stands.

**Review by ndaelman-hu — COMMENTED** [review]
This is okay for a quick fix. If I could take a look at the error or the raw files, I could probabaly hunt down the real reason.
In the past, there were issues matching the basis set specification due to Gaussian's formatting.

**Comment by ndaelman-hu** [comment]
@ladinesa Okay, I got access to the failing upload. You made the right call: there simply isn't any basis set mentioned.
Likely, this is due to:

 1. this calculation starting from a previous SCF (not uploaded).
 2. this is a frequency calculation, with a different emphasis.
 
 
Still, this is pretty bad practice on Gaussian's side, as SCF routines shouldn't tacitly assume the basis set.
Fro the long-term, I'll see if I can hunt down their default, though Gaussian is likely to have some complex logic in assigning it.

**Comment by ndaelman-hu** [comment]
@JosePizarro3 Regarding the upload you linked, all the Gaussian files parsed well there.
The issue seems to be with Orca, as we don't support CASSCF, yet. I have this on the planned list, but quite low.

We either add a schema there, or we harden the parsing not to fail for unrecognized methods.

**Comment by ndaelman-hu** [comment]
Digging further, the default is [STO-3G](https://gaussian.com/basissets/). I can see to improve on the current correction then.

**Comment by ladinesa** [comment]
@ndaelman there is a problematic test. is it possible for you to look at it so we can merge it already? thanks

**Comment by ndaelman-hu** [comment]
So, some final comments here that will lead to new issues:
1. we need to add missing basis set specs to our tests
2. we will have to extend "GEN" (or "Gen") parsing, as this indicates a user-defined basis set

**Pull request merged**